src/blocks/BlockIO.as

```diff
@@ -146,15 +146,15 @@ public class BlockIO {
 		return [spec, undefinedBlockType, 0, op]; // no match found
 	}
 
-	private static function argsForCmd(cmd:Array, reverseArgs:Boolean):Array {
+	private static function argsForCmd(cmd:Array, numArgs:uint, reverseArgs:Boolean):Array {
 		// Return an array of zero or more arguments for the given command.
-		// Skip substacks. Arguments may be literal values or reporter blocks (expressions).
+		// Arguments may be literal values or reporter blocks (expressions).
 		var result:Array = [];
-		for (var i:int = 1; i < cmd.length; i++) {
+		for (var i:int = 1; i < 1 + numArgs; i++) {
```

Comment:
Maybe we could increment numArgs before entering the loop. Or, the argument could be renamed to cmdSize or something similar.